### PR TITLE
Sanitation

### PR DESCRIPTION
**Pull request by satviknarang8**
## Describe your changes
1. SignupScreen.tsx:
- Added email and password validation.
- Displayed error messages for invalid inputs and existing accounts.
- Disabled signup button for invalid inputs or if terms of service are not agreed.

2. LoginScreen.tsx:
- Displayed error message for incorrect email or password.

3. SignupButtonFromSignup.tsx:
- Added disabled prop and conditional styling for the disabled state.

4. Auth.ts:
- Updated signupWithEmail and loginWithEmail to return specific error messages and success states.

## Issue ticket number and link
#30 

## Checklist
- [x] I have performed a self-review of my code
- [x] I have used `npx expo-doctor` to review any dependency issues.
- [x] I have pulled and merged any recent changes from the main branch (no merge conflicts)
- [x] I have tested thoroughly any edge cases that might break the application
- [x] I have deleted any console.log statements and useless comments

### CODE REVIEW

File: components/__tests__/__snapshots__/StyledText-test.js.snap
Comment (dave-cao) [inline components/__tests__/__snapshots__/StyledText-test.js.snap:1]: What is this file for?

File: app.json [hunk 1]
```diff
@@ -16,7 +16,8 @@
       "supportsTablet": true,
       "associatedDomains": [
         "applinks:firmcollective.org"
-      ]
+      ],
+      "bundleIdentifier": "com.firmcollective.OneVoiceEcho"
```
Comment (dave-cao) [inline app.json:20]: What is the change here for?

File: components/auth/LoginScreen.tsx
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:1]: When I run this on android, and try to go to the login page I get this error:
image

File: components/auth/LoginScreen.tsx [hunk 2]
```diff
@@ -47,16 +54,16 @@ export default function LoginScreen() {
           <View>
             <TextInput
               style={styles.textInput}
-              onChangeText={(text: string) => setEmail(text)}
+              onChangeText={(text: string) => setEmail(text)} // Update email state
```
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:57]: redundant comment, not needed.

File: components/auth/LoginScreen.tsx [hunk 3]
```diff
@@ -47,16 +54,16 @@ export default function LoginScreen() {
...
               placeholder="Email"
             />
           </View>
           <View>
             <TextInput
               style={styles.textInput}
-              secureTextEntry={!showPassword}
-              onChangeText={(text: string) => setPassword(text)}
+              secureTextEntry={!showPassword} // Conditionally render password visibility
+              onChangeText={(text: string) => setPassword(text)} // Update password state
```
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:66]: redundant comment, not needed.

File: components/auth/LoginScreen.tsx [hunk 4]
```diff
@@ -67,7 +74,7 @@ export default function LoginScreen() {
                 right: 10,
                 padding: 5,
               }}
-              onPress={togglePasswordVisibility}
+              onPress={togglePasswordVisibility} // Toggle password visibility
```
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:77]: Redundant comment, not needed.

File: components/auth/LoginScreen.tsx [hunk 5]
```diff
@@ -180,13 +191,11 @@ const styles = StyleSheet.create({
...
   },
-  textBottom: {
-    marginTop: 120,
-    fontSize: 12,
-  },
-  textBottom2: {
-    marginTop: 180,
-    fontSize: 12,
+  errorText: {
+    color: 'red',
```
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:195]: from the figma file, we want the color to be #A82424, not just red.

File: components/auth/LoginScreen.tsx
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:1]: You also need to add when the login button should be disabled. I only saw it for the signup button.

File: components/auth/SignupScreen.tsx
Comment (dave-cao) [inline components/auth/SignupScreen.tsx:1]: Bug: After an error message, the signup button is still disabled
image

Bug: After successful email and clicking up acknowledge, a single character in password should not enable signup button
image

Bug: Why is the password visibility toggle on the email input field and not the password input field?
image

**Review by dave-cao — CHANGES_REQUESTED** [review]
There are still quite a few bugs that need to be addressed. The login page crashes on first load. I have more detailed comments above.

File: components/auth/LoginScreen.tsx
Comment (dave-cao) [inline components/auth/LoginScreen.tsx:1]: I am still getting the same error when I try to access the LoginScreen
image

**Review by dave-cao — CHANGES_REQUESTED** [review]
The SignupScreen looks really good! I tested it out and everything is working as expected. For some reason, I am still getting the same error when I try to access the LoginScreen. I included a picture in an above comment.

You also need to rebase the updates from main into this branch before we can merge! 

It looks like its almost done!!
